polyfill/lib/ecmascript.mjs

```diff
@@ -1290,10 +1290,10 @@ export const ES = ObjectAssign({}, ES2020, {
     ES.RejectDate(referenceISOYear, month, day);
     if (calendar === undefined) calendar = ES.GetISO8601Calendar();
     calendar = ES.ToTemporalCalendar(calendar);
-    if (referenceISOYear === undefined) referenceISOYear = 1972;
-    let result = new constructor(month, day, calendar, referenceISOYear);
-    if (!ES.IsTemporalMonthDay(result)) throw new TypeError('invalid result');
-    return result;
+
+    const PlainMonthDay = GetIntrinsic('%Temporal.PlainMonthDay%');
+    let result = new PlainMonthDay(month, day, calendar, referenceISOYear);
```

Comment:
In the vast majority of cases, calendar will be ISO and referenceISOYear will not be included in the string. In that case, is the double-construction needed?  Seems like a worthwhile optimization to only construct twice if there's a year in the input.

Also (out of scope for this PR?), is there a case where we'll have a string like `09-04[c=japanese]` where the year is omitted? Seems like that format should be disallowed; if the calendar is in the string, then a year should be required.

Reply:
I agree it's weird, and does add more observable calls to methods of `calendar`. I'll take a look at this tomorrow.

We do already disallow `mm-dd[calendar]`.
Thanks for taking a look!